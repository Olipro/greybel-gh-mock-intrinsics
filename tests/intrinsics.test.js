const { Interpreter, Debugger, CustomFunction } = require('greybel-interpreter');
const defaultInit = require('greybel-intrinsics').init;
const { init } = require('../dist');
const fs = require('fs');
const path = require('path');
const testFolder = path.resolve(__dirname, 'scripts');

let printMock;
const pseudoAPI = new Map();

const testDate = new Date(1642924301240);

Date.now = () => testDate.getTime();

pseudoAPI.set(
  'print',
  CustomFunction.createExternal('print', (fnCtx, self, args) => {
    // console.log(args);
    printMock(args.get('value').toString());
  }).addArgument('value')
);

class TestDebugger extends Debugger {
  debug() {}
}

describe('interpreter', function () {
  beforeEach(function () {
    printMock = jest.fn();
  });

  describe('default scripts', function () {
    fs.readdirSync(testFolder).forEach((file) => {
      const filepath = path.resolve(testFolder, file);

      test(path.basename(filepath), async () => {
        const interpreter = new Interpreter({
          target: filepath,
          debugger: new TestDebugger()
        });
        let success = false;

        pseudoAPI.set(
          'exit',
          CustomFunction.createExternal('exit', (fnCtx, self, args) => {
            interpreter.exit();
          })
        );

        interpreter.setApi(defaultInit(init(pseudoAPI)));

        try {
          await interpreter.run();
          success = true;
        } catch (e) {
          console.log(`${filepath} failed with: `, e);
        }

        expect(success).toEqual(true);
        for (const call of printMock.mock.calls) {
          expect(call[0]).toMatchSnapshot();
        }
      });
    });
  });
});
