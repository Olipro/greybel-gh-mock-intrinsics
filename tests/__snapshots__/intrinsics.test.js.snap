// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interpreter default scripts blockchain.src 1`] = `"blockchain"`;

exports[`interpreter default scripts blockchain.src 2`] = `"blockchain"`;

exports[`interpreter default scripts blockchain.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 6`] = `"coin"`;

exports[`interpreter default scripts blockchain.src 7`] = `"wallet"`;

exports[`interpreter default scripts blockchain.src 8`] = `"wallet"`;

exports[`interpreter default scripts blockchain.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 1`] = `"coin"`;

exports[`interpreter default scripts coin.src 2`] = `"coin"`;

exports[`interpreter default scripts coin.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 9`] = `"subWallet"`;

exports[`interpreter default scripts coin.src 10`] = `"[ subWallet ]"`;

exports[`interpreter default scripts coin.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 12`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 13`] = `"Not yet supported"`;

exports[`interpreter default scripts crypto.src 1`] = `"crypto"`;

exports[`interpreter default scripts crypto.src 2`] = `"crypto"`;

exports[`interpreter default scripts crypto.src 3`] = `
"Waiting for beacon frame (BSSID: bssid-test-uuid)
Sending Authentication Request (Open system) [ACK]
Authentication succesful
Sending Association Request
Association succesful :-)"
`;

exports[`interpreter default scripts crypto.src 4`] = `"0/25000"`;

exports[`interpreter default scripts crypto.src 5`] = `"728/25000"`;

exports[`interpreter default scripts crypto.src 6`] = `""`;

exports[`interpreter default scripts crypto.src 7`] = `"false"`;

exports[`interpreter default scripts crypto.src 8`] = `"Key not found. More acks needed"`;

exports[`interpreter default scripts crypto.src 9`] = `""`;

exports[`interpreter default scripts crypto.src 10`] = `"shallnotpass"`;

exports[`interpreter default scripts crypto.src 11`] = `""`;

exports[`interpreter default scripts generics.src 1`] = `""`;

exports[`interpreter default scripts generics.src 2`] = `"098f6bcd4621d373cade4e832627b4f6"`;

exports[`interpreter default scripts generics.src 3`] = `"1642924301240"`;

exports[`interpreter default scripts generics.src 4`] = `"142.567.134.56"`;

exports[`interpreter default scripts generics.src 5`] = `
"Domain name: metaldyne.org
Administrative contact: Hercule Beane
Email address: mrpacker2003@metaldyne.org
Phone: 123456891"
`;

exports[`interpreter default scripts generics.src 6`] = `"true"`;

exports[`interpreter default scripts generics.src 7`] = `"true"`;

exports[`interpreter default scripts generics.src 8`] = `"COMMAND"`;

exports[`interpreter default scripts generics.src 9`] = `"/home"`;

exports[`interpreter default scripts generics.src 10`] = `"/home/test"`;

exports[`interpreter default scripts generics.src 11`] = `"/home/test/myprogramm"`;

exports[`interpreter default scripts generics.src 12`] = `"test"`;

exports[`interpreter default scripts generics.src 13`] = `"test@briggs.de"`;

exports[`interpreter default scripts generics.src 14`] = `"s1jbeulm"`;

exports[`interpreter default scripts generics.src 15`] = `"clearing screen"`;

exports[`interpreter default scripts generics.src 16`] = `""`;

exports[`interpreter default scripts generics.src 17`] = `"/home/test/myprogramm"`;

exports[`interpreter default scripts generics.src 18`] = `
"COL1    COL2    COL3    
one     two     three   
thing1  thing2  thing3  "
`;

exports[`interpreter default scripts generics.src 19`] = `"test"`;

exports[`interpreter default scripts generics.src 20`] = `"test"`;

exports[`interpreter default scripts generics.src 21`] = `"test"`;

exports[`interpreter default scripts generics.src 22`] = `"test"`;

exports[`interpreter default scripts generics.src 23`] = `"test"`;

exports[`interpreter default scripts generics.src 24`] = `
"
"
`;

exports[`interpreter default scripts ls.src 1`] = `
"Desktop     
Downloads   
Config      
myprogramm  "
`;

exports[`interpreter default scripts meta-mail.src 1`] = `"metaMail"`;

exports[`interpreter default scripts meta-mail.src 2`] = `"metaMail"`;

exports[`interpreter default scripts meta-mail.src 3`] = `"[  ]"`;

exports[`interpreter default scripts meta-mail.src 4`] = `"Mail not found"`;

exports[`interpreter default scripts meta-mail.src 5`] = `"true"`;

exports[`interpreter default scripts meta-mail.src 6`] = `
"[ 

MailID: 5171bca0-1b95-4188-8453-46a19c6bb2d2
From: test@test.org
Subject: hello
dude ]"
`;

exports[`interpreter default scripts meta-mail.src 7`] = `"5171bca0-1b95-4188-8453-46a19c6bb2d2"`;

exports[`interpreter default scripts meta-mail.src 8`] = `
"
From: test@test.org
Subject: hello
dude

"
`;

exports[`interpreter default scripts meta-mail.src 9`] = `"true"`;

exports[`interpreter default scripts meta-mail.src 10`] = `"[  ]"`;

exports[`interpreter default scripts metaxploit.src 1`] = `"metaxploit"`;

exports[`interpreter default scripts metaxploit.src 2`] = `"metaxploit"`;

exports[`interpreter default scripts metaxploit.src 3`] = `""`;

exports[`interpreter default scripts metaxploit.src 4`] = `""`;

exports[`interpreter default scripts metaxploit.src 5`] = `"error: service rshelld is not running"`;

exports[`interpreter default scripts metaxploit.src 6`] = `"metaLib"`;

exports[`interpreter default scripts metaxploit.src 7`] = `"metaLib"`;

exports[`interpreter default scripts metaxploit.src 8`] = `"[ 0xC8CE05AB, 0xFDC63532, 0xDC4BD50C ]"`;

exports[`interpreter default scripts metaxploit.src 9`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>_tr_flush_blockblock_start</b>. Buffer overflow.
Unsafe check: loop in array <b>hash_mask</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0.0</b>
* Checking an active user."
`;

exports[`interpreter default scripts metaxploit.src 10`] = `"crypto"`;

exports[`interpreter default scripts metaxploit.src 11`] = `"1.0.0.0"`;

exports[`interpreter default scripts metaxploit.src 12`] = `"[ 0xC8CE05AB, 0xFDC63532, 0xDC4BD50C ]"`;

exports[`interpreter default scripts metaxploit.src 13`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>_tr_flush_blockblock_start</b>. Buffer overflow.
Unsafe check: loop in array <b>hash_mask</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0.0</b>
* Checking an active user."
`;

exports[`interpreter default scripts metaxploit.src 14`] = `
"
Starting attack...success!
Privileges obtained from user: test"
`;

exports[`interpreter default scripts metaxploit.src 15`] = `"shell"`;

exports[`interpreter default scripts metaxploit.src 16`] = `
"
Starting attack...success!
Privileges obtained from user: test"
`;

exports[`interpreter default scripts metaxploit.src 17`] = `"shell"`;

exports[`interpreter default scripts metaxploit.src 18`] = `"netSession"`;

exports[`interpreter default scripts metaxploit.src 19`] = `"netSession"`;

exports[`interpreter default scripts metaxploit.src 20`] = `"metaLib"`;

exports[`interpreter default scripts metaxploit.src 21`] = `"metaLib"`;

exports[`interpreter default scripts metaxploit.src 22`] = `"ssh"`;

exports[`interpreter default scripts metaxploit.src 23`] = `"1.0.0.0"`;

exports[`interpreter default scripts metaxploit.src 24`] = `"[ 0xA12E2B71, 0xE2D91188, 0x3089D743 ]"`;

exports[`interpreter default scripts metaxploit.src 25`] = `
"decompiling source...
searching unsecure values...
Unsafe check: string copy in <b>void</b>. Buffer overflow.
* Using namespace <b>net.so</b> compiled at version >= <b>1.0.0.0</b>
* Checking registered users equal to 2.
Unsafe check: loop in array <b>DeleteLinei</b>. Buffer overflow.
* Using namespace <b>init.so</b> compiled at version >= <b>1.0.0.0</b>
* Checking registered users equal to 4.
* Checking an active user.
* Checking root active user.
* 4 port forwarding configured from router to the target computer.
Unsafe check: string copy in <b>treeDistancesdyn_tree</b>. Buffer overflow.
* 3 port forwarding configured from router to the target computer.
Unsafe check: loop in array <b>match_length</b>. Buffer overflow.
Unsafe check: string copy in <b>ifrestoreWindow</b>. Buffer overflow.
* Using namespace <b>init.so</b> compiled at version >= <b>1.0.0.0</b>
* Checking an active user.
* 3 port forwarding configured from router to the target computer."
`;

exports[`interpreter default scripts metaxploit.src 26`] = `
"
Starting attack...success!
Privileges obtained from user: guest"
`;

exports[`interpreter default scripts metaxploit.src 27`] = `"file"`;

exports[`interpreter default scripts metaxploit.src 28`] = `"kernel_router"`;

exports[`interpreter default scripts metaxploit.src 29`] = `"1.0.0.0"`;

exports[`interpreter default scripts metaxploit.src 30`] = `"[ 0xB1AFF995, 0x2795D2B7, 0x95903F51, 0xC829866A ]"`;

exports[`interpreter default scripts metaxploit.src 31`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>linesInsertedOrRemoved</b>. Buffer overflow.
Unsafe check: string copy in <b>distance</b>. Buffer overflow.
Unsafe check: loop in array <b>QueueToken</b>. Buffer overflow.
Unsafe check: string copy in <b>Title</b>. Buffer overflow.
Unsafe check: string copy in <b>ALessThanB</b>. Buffer overflow."
`;

exports[`interpreter default scripts metaxploit.src 32`] = `
"
Starting attack...success!
Accessing firewall config...
Firewall disabled. (Firewalls are not yet supported in greybel!)"
`;

exports[`interpreter default scripts metaxploit.src 33`] = `"true"`;

exports[`interpreter default scripts mkdir.src 1`] = `"false"`;

exports[`interpreter default scripts router.src 1`] = `"router"`;

exports[`interpreter default scripts router.src 2`] = `"router"`;

exports[`interpreter default scripts router.src 3`] = `"router"`;

exports[`interpreter default scripts router.src 4`] = `"router"`;

exports[`interpreter default scripts router.src 5`] = `"142.32.54.56"`;

exports[`interpreter default scripts router.src 6`] = `"192.168.0.10"`;

exports[`interpreter default scripts router.src 7`] = `"port"`;

exports[`interpreter default scripts router.src 8`] = `"ssh 1.0.0.0"`;

exports[`interpreter default scripts router.src 9`] = `"[  ]"`;

exports[`interpreter default scripts scripts.src 1`] = `
"Desktop    
Downloads  
Config     "
`;

exports[`interpreter default scripts scripts.src 2`] = `
"-r-x--x--x  root  root  345   00:00  build             
-r-x--x--x  root  root  262   00:00  cat               
-r-x--x--x  root  root  848   00:00  cp                
-r-x--x--x  root  root  261   00:00  kill              
-r-x--x--x  root  root  1014  00:00  ls                
-r-x--x--x  root  root  505   00:00  mkdir             
-r-x--x--x  root  root  918   00:00  mv                
-r-x--x--x  root  root  89    00:00  ps                
-r-x--x--x  root  root  23    00:00  pwd               
-r-x--x--x  root  root  401   00:00  rm                
-r-x--x--x  root  root  637   00:00  sudo              
-r-x--x--x  root  root  373   00:00  touch             
-r-x--x--x  root  root  24    00:00  whoami            
-r-x--x--x  root  root  415   00:00  aircrack          
-r-x--x--x  root  root  249   00:00  aireplay          
-r-x--x--x  root  root  470   00:00  airmon            
-r-x--x--x  root  root  2157  00:00  apt-get           
-r-x--x--x  root  root  343   00:00  ftp               
-r-x--x--x  root  root  734   00:00  ifconfig          
-r-x--x--x  root  root  298   00:00  iwconfig          
-r-x--x--x  root  root  388   00:00  iwlist            
-r-x--x--x  root  root  779   00:00  nmap              
-r-x--x--x  root  root  126   00:00  nslookup          
-r-x--x--x  root  root  196   00:00  ping              
-r-x--x--x  root  root  612   00:00  rshell-interface  
-r-x--x--x  root  root  442   00:00  smtp-user-list    
-r-x--x--x  root  root  354   00:00  sniffer           
-r-x--x--x  root  root  394   00:00  ssh               
-r-x--x--x  root  root  113   00:00  whois             
-r-x--x--x  root  root  284   00:00  chgrp             
-r-x--x--x  root  root  288   00:00  chmod             
-r-x--x--x  root  root  284   00:00  chown             
-r-x--x--x  root  root  233   00:00  groupadd          
-r-x--x--x  root  root  236   00:00  groupdel          
-r-x--x--x  root  root  159   00:00  groups            
-r-x--x--x  root  root  266   00:00  passwd            
-r-x--x--x  root  root  268   00:00  useradd           
-r-x--x--x  root  root  269   00:00  userdel           
-r-x--x--x  root  root  998   00:00  decipher          
-r-x--x--x  root  root  693   00:00  scanlib           
-r-x--x--x  root  root  519   00:00  scanrouter        "
`;

exports[`interpreter default scripts scripts.src 3`] = `
"USER  PID   CPU  MEM    COMMAND             
root  2705  8.5  25.02  kernel_task         
root  7499  9.2  23.05  Xorg                "
`;

exports[`interpreter default scripts scripts.src 4`] = `"/root"`;

exports[`interpreter default scripts scripts.src 5`] = `
"
Connected to Wi-Fi:
Essid: essid-test-uuid
Bssid: bssid-test-uuid
----------------
Public IP: 142.32.54.56
Local IP: 192.168.253.21
Gateway: 192.168.0.10
"
`;

exports[`interpreter default scripts service.src 1`] = `"service"`;

exports[`interpreter default scripts service.src 2`] = `"service"`;

exports[`interpreter default scripts service.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts service.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts service.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts shell.src 1`] = `"shell"`;

exports[`interpreter default scripts shell.src 2`] = `"computer"`;

exports[`interpreter default scripts shell.src 3`] = `"computer"`;

exports[`interpreter default scripts shell.src 4`] = `"[ port ]"`;

exports[`interpreter default scripts shell.src 5`] = `"Can't create folder /root/foo. Permission denied"`;

exports[`interpreter default scripts shell.src 6`] = `"true"`;

exports[`interpreter default scripts shell.src 7`] = `"file"`;

exports[`interpreter default scripts shell.src 8`] = `"true"`;

exports[`interpreter default scripts shell.src 9`] = `"Can't create file /root/foo-file. Permission denied"`;

exports[`interpreter default scripts shell.src 10`] = `"true"`;

exports[`interpreter default scripts shell.src 11`] = `"false"`;

exports[`interpreter default scripts shell.src 12`] = `
"USER  PID   CPU  MEM    COMMAND      
root  5539  7.7  17.86  kernel_task  
root  379   8.2  23.74  Xorg         "
`;

exports[`interpreter default scripts shell.src 13`] = `"wlan0 ggrdk4zo true"`;

exports[`interpreter default scripts shell.src 14`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 15`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 16`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 17`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 18`] = `""`;

exports[`interpreter default scripts shell.src 19`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 20`] = `"false"`;

exports[`interpreter default scripts shell.src 21`] = `"[ 44:8E:73:9E:F2:63 100% circinus, 23:B9:F1:05:1C:D9 57% apricot_rain, 76:FF:BA:97:28:4C 62% nisse, 6D:63:9B:5D:1A:61 68% driest, F2:8C:C2:3E:E6:6F 51% mynameisskiller ]"`;

exports[`interpreter default scripts shell.src 22`] = `"true"`;

exports[`interpreter default scripts shell.src 23`] = `"Error: Invalid IP address"`;

exports[`interpreter default scripts shell.src 24`] = `"192.168.0.10"`;

exports[`interpreter default scripts shell.src 25`] = `"WIFI"`;

exports[`interpreter default scripts shell.src 26`] = `"192.168.189.79"`;

exports[`interpreter default scripts shell.src 27`] = `"142.32.54.56"`;

exports[`interpreter default scripts shell.src 28`] = `"file"`;

exports[`interpreter default scripts shell.src 29`] = `"file"`;

exports[`interpreter default scripts shell.src 30`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 31`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 32`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 33`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 34`] = `"/root"`;

exports[`interpreter default scripts shell.src 35`] = `"file"`;

exports[`interpreter default scripts shell.src 36`] = `"root"`;

exports[`interpreter default scripts shell.src 37`] = `"can't open /root Binary file."`;

exports[`interpreter default scripts shell.src 38`] = `"can't open /root Binary file."`;

exports[`interpreter default scripts shell.src 39`] = `"true"`;

exports[`interpreter default scripts shell.src 40`] = `"true"`;

exports[`interpreter default scripts shell.src 41`] = `"false"`;

exports[`interpreter default scripts shell.src 42`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 43`] = `"[ file, file, file, file ]"`;

exports[`interpreter default scripts shell.src 44`] = `"[  ]"`;

exports[`interpreter default scripts shell.src 45`] = `"root"`;

exports[`interpreter default scripts shell.src 46`] = `"Permission denied"`;

exports[`interpreter default scripts shell.src 47`] = `"root"`;

exports[`interpreter default scripts shell.src 48`] = `"Permission denied"`;

exports[`interpreter default scripts shell.src 49`] = `"0"`;

exports[`interpreter default scripts shell.src 50`] = `"false"`;

exports[`interpreter default scripts shell.src 51`] = `"computer"`;

exports[`interpreter default scripts shell.src 52`] = `"false"`;

exports[`interpreter default scripts shell.src 53`] = `"true"`;

exports[`interpreter default scripts shell.src 54`] = `"file"`;

exports[`interpreter default scripts shell.src 55`] = `"true"`;

exports[`interpreter default scripts shell.src 56`] = `"file"`;

exports[`interpreter default scripts shell.src 57`] = `""`;

exports[`interpreter default scripts shell.src 58`] = `"123"`;

exports[`interpreter default scripts shell.src 59`] = `"true"`;

exports[`interpreter default scripts shell.src 60`] = `"true"`;

exports[`interpreter default scripts shell.src 61`] = `"testg"`;

exports[`interpreter default scripts shell.src 62`] = `"true"`;

exports[`interpreter default scripts shell.src 63`] = `"true"`;

exports[`interpreter default scripts shell.src 64`] = `"true"`;

exports[`interpreter default scripts shell.src 65`] = `"true"`;

exports[`interpreter default scripts shell.src 66`] = `"true"`;

exports[`interpreter default scripts shell.src 67`] = `"file"`;

exports[`interpreter default scripts shell.src 68`] = `""`;

exports[`interpreter default scripts shell.src 69`] = `"drwxr--rwx"`;

exports[`interpreter default scripts shell.src 70`] = `"true"`;

exports[`interpreter default scripts shell.src 71`] = `"true"`;

exports[`interpreter default scripts shell.src 72`] = `"i am admin"`;

exports[`interpreter default scripts shell.src 73`] = `"root"`;

exports[`interpreter default scripts shell.src 74`] = `""`;

exports[`interpreter default scripts shell.src 75`] = `"test"`;

exports[`interpreter default scripts shell.src 76`] = `"root"`;

exports[`interpreter default scripts shell.src 77`] = `""`;

exports[`interpreter default scripts shell.src 78`] = `"test-here"`;

exports[`interpreter default scripts sub-wallet.src 1`] = `"subWallet"`;

exports[`interpreter default scripts sub-wallet.src 2`] = `"subWallet"`;

exports[`interpreter default scripts sub-wallet.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 10`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 1`] = `"wallet"`;

exports[`interpreter default scripts wallet.src 2`] = `"wallet"`;

exports[`interpreter default scripts wallet.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 10`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 12`] = `"Not yet supported"`;
